### ISSUE TITLE
TypeError using by_distance on sqlite3

### ISSUE THREAD

**Issue by kmmndr** [geokit/geokit-rails#155]
Calling `by_distance` scope on sqlite3 produce the following error

```
TypeError (no implicit conversion of Symbol into Integer):

sqlite3 (1.4.2) lib/sqlite3/database.rb:396:in `define_function_with_flags'
sqlite3 (1.4.2) lib/sqlite3/database.rb:396:in `create_function'
geokit-rails (2.3.1) lib/geokit-rails/adapters/sqlite.rb:6:in `add_numeric'
geokit-rails (2.3.1) lib/geokit-rails/adapters/sqlite.rb:13:in `add_math'
geokit-rails (2.3.1) lib/geokit-rails/adapters/sqlite.rb:22:in `load'
geokit-rails (2.3.1) lib/geokit-rails/acts_as_mappable.rb:111:in `geokit_finder_adapter'
geokit-rails (2.3.1) lib/geokit-rails/acts_as_mappable.rb:345:in `sphere_distance_sql'
geokit-rails (2.3.1) lib/geokit-rails/acts_as_mappable.rb:215:in `distance_sql'
geokit-rails (2.3.1) lib/geokit-rails/acts_as_mappable.rb:155:in `by_distance' 
```

**Comment by kmmndr** [comment]
It seems to be between [geokit-rails](https://github.com/geokit/geokit-rails/blob/cc5fd43ab4e69878fb31ebd1fc22918e2952b560/lib/geokit-rails/adapters/sqlite.rb#L6) and [sqlite3-ruby](https://github.com/sparklemotion/sqlite3-ruby/blob/6fc5073ecc9c90d7458a43d726e92201e4c3ff65/lib/sqlite3/database.rb#L395) as the former is sending a symbol as third argument and the later expecting the third argument to be found as integer in an enum

```ruby
# lib/geokit-rails/adapters/sqlite.rb
      def self.add_numeric(name) 
        @@connection.create_function name, 1, :numeric do |func, *args|
          func.result = yield(*args)
        end
      end
```

```ruby
# lib/sqlite3/database.rb
    def create_function name, arity, text_rep=Constants::TextRep::UTF8, &block
      define_function_with_flags(name, text_rep) do |*args|
        fp = FunctionProxy.new
        block.call(fp, *args)
        fp.result
      end
      self
    end
```

**Comment by kmmndr** [comment]
Removing the third argument (and keeping `UTF8` by default) works as expected

```patch
diff --git a/lib/geokit-rails/adapters/sqlite.rb b/lib/geokit-rails/adapters/sqlite.rb
index 1042eff..ae1754c 100644
--- a/lib/geokit-rails/adapters/sqlite.rb
+++ b/lib/geokit-rails/adapters/sqlite.rb
@@ -3,7 +3,7 @@ module Geokit
     class SQLite < Abstract
       
       def self.add_numeric(name) 
-        @@connection.create_function name, 1, :numeric do |func, *args|
+        @@connection.create_function name, 1 do |func, *args|
           func.result = yield(*args)
         end
       end
```

**Comment by paulogeyer** [comment]
+1

I have the same problem here, this patch is going to be added to the master?

**Comment by ryankopf** [comment]
This is fixed in 2.3.2 now.
